product/scanning.py

```diff
@@ -22,16 +22,50 @@
     AnalysisResult,
     TriageScope,
 )
+from concourse.model.base import (
+    AttribSpecMixin,
+    AttributeSpec,
+)
 from util import not_none, warning, check_type, info, urljoin
 from container.registry import retrieve_container_image, publish_container_image
 from .model import ContainerImage, Component, UploadResult, UploadStatus
 
 
-class ProcessingMode(Enum):
+
+class ProcessingMode(AttribSpecMixin, Enum):
     UPLOAD_IF_CHANGED = 'upload_if_changed'
     RESCAN = 'rescan'
     FORCE_UPLOAD = 'force_upload'
 
+    @classmethod
+    def _attribute_specs(cls):
+        return (
+            AttributeSpec.optional(
+                name=cls.UPLOAD_IF_CHANGED.value,
+                default=True,
```

Comment:
i'm still not happy with the mis-use of the 'default' attribute for enums

Reply:
Repalced with a lesser mis-use. Probably worth discussing further tomorrow morning.

Reply:
After another look: I guess it's actually used as intended now.